### PR TITLE
Fixer for typedef rule with variable declarations turned on

### PR DESCRIPTION
**Pull request by WA9ACE**
Fixes variable declarations missing type annotations when using

```json
"@typescript-eslint/typedef": ["error", {
  "variableDeclaration": true
}]
```

### CODE REVIEW

**Comment by bradzacher** [comment]
Thanks for the contribution! It's always a good idea to raise an issue before a PR, so we can discuss and make sure it's something we want to include in this project.

A few reasons this won't be merged:

1. 
This rule is intentionally simple, and doesn't use type information on purpose. Using type information has an overhead, which many of our users do not want to incur. So by introducing type information (a) it is a breaking change (because people that didn't have their config setup for type info will no longer be able to use the rule), and (b) it restricts the rule, and prevents many users from using it.

In this case, we don't want to introduce type information into this rule.  
The general rule for this plugin is "don't use type information unless you have to".

2.
An automatic fixer for this sort of rule is not a good idea.  
Sure, it makes it convenient to correct mistakes. However, it also contravenes the point of the rule; it means you don't have to think about the types of your variables. There's not really any point to a rule which just adds type annotations based on the inferred type.

This is the same reason we don't have a fixer for our `explicit-member-accessibility` rule. It'd be super simple to default it to `public`, but that goes against the point of the linter telling devs to think about the accessibility they're giving.

3.
You might have weird cases with this where you get an overly verbose, or incorrect type placed in. EG `const x = { a: 'a' }` would fix to `const x: { a: string } = { a: 'a' }`, which would be problematic for larger object types. It's not feasible to "detect" all types in the current scope and find one that matches (nor would it be necessarily correct).

I could see cases where you might incorrectly widen from an exact string/number ('a', 1, etc) type to the general string/number type, which could break code.

**Comment by WA9ACE** [comment]
No worries, though personally I like having full type information inline even if it can be inferred I understand the reasons not to include this, and I'm definitely a very small minority that wants this. Is there some documentation you could point me to for writing a typescript-eslint plugin that integrates well with this package's infrastructure for writing rules?

**Comment by bradzacher** [comment]
No docs yet. It's on the todo list. Its still a pretty niche thing, so we haven't gotten around to it yet.

Tldr would be: Copy the same things we do for `eslint-plugin`, and you should do fine.
